### PR TITLE
Update Node to 16

### PR DESCRIPTION
**Pull request by DerekRoberts**
GitHub Actions has deprecated Node 12 in favour of 16, the new LTS.  Their Actions are expected to be moved by summer 2023, so we have lots of time!

I'm unfortunately not sure how to test this, so please don't assume the PR is pure genius.

GitHub blog release
https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12

Sample deprecation warning:
`Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: zaproxy/action-full-scan`

### CODE REVIEW

**Comment by thc202** [comment]
DCO check is failing: https://github.com/zaproxy/action-full-scan/pull/44/checks?check_run_id=8949954101

**Comment by DerekRoberts** [comment]
> DCO check is failing: https://github.com/zaproxy/action-full-scan/pull/44/checks?check_run_id=8949954101

Better?
https://github.com/zaproxy/action-full-scan/pull/44/checks?check_run_id=8963152958

**Comment by DerekRoberts** [comment]
@the202 Changes to .github/workflows/check-dist.yml pulled out.  Hopefully that'll lead to a pass for `Compare the expected and actual dist/directories`.

**Comment by thc202** [comment]
All should be using 16, you have to update dist.

**Comment by thc202** [comment]
The changelog should be updated.

File: .github/workflows/check-dist.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ on:
     branches:
       - master
     paths-ignore:
-      - '**.md'
+      - "**.md"
   pull_request:
     paths-ignore:
-      - '**.md'
+      - "**.md"
```
Comment (thc202) [inline .github/workflows/check-dist.yml:16]: Revert.

File: .github/workflows/check-dist.yml [hunk 2]
```diff
@@ -23,10 +23,10 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set Node.js 12.x
+      - name: Set Node.js 16.x
```
Comment (thc202) [inline .github/workflows/check-dist.yml:26]: Remove the version instead.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+## [0.5.0] - 2022-10-21
```
Comment (thc202) [inline CHANGELOG.md:8]: Remove.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+### Changed
+- Updated GitHub Action and workflow for Node 16.
```
Comment (thc202) [inline CHANGELOG.md:10]: This is not necessary, this does not affect the released action itself.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+
+### Fixed
+- Addresses GitHub Actions Node 12 deprecation.
```
Comment (thc202) [inline CHANGELOG.md:13]: This is not a bug, should be under changed.

**Comment by DerekRoberts** [comment]
@thc202 Requests completed.

**Comment by DerekRoberts** [comment]
@thc202 Anything else I can do to help this PR along?

**Comment by thc202** [comment]
Changes look good, commits should be squashed.

**Comment by DerekRoberts** [comment]
> Changes look good, commits should be squashed.

Done!

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Comment by gil0109** [comment]
Can this included in version 0.4.0?  My git-actions is still warning me and I believe 0.4.0 is the latest version i can use.  Thank you.

**Comment by thc202** [comment]
v0.4.0 was already released, this will be included in the next version/release.

**Pull request merged**